### PR TITLE
Added CSequenceMain and CGameParameter to CApplicationDocument and CGameParameter mapping

### PR DESCRIPTION
**Pull request by NextinMono**
Hello again, I tried to follow the naming scheme as best as I could.
This PR contains 2 changes for CApplicationDocument with the locations of CSequenceMainImpl and CGameParameter, the latter also has been mostly mapped out.
The name "SStageParameters" is a made up name as both structs in CGameParameter don't have any name, I've called it this as it contains certain variables which are related to stages, such as the archive name of the stage terrain.
SField7C of CGameParameter contains a couple of arrays of an unknown type, so temporarily they have been replaced with padding to keep alignment; m_Field14 of SStageParameters also has an unknown type, although I have confirmed it is a list

### CODE REVIEW

File: Sonic/System/GameParameter.h [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			int m_Field1A4;
+			int m_Field1A8;
+			int m_Field1AC;
+			Hedgehog::Base::CSharedString m_BGMCueNodeName;
+			int m_Field1B4;
+			int m_Field1B8;
+			int m_Field1BC;
+
+		};
+		SField7C* m_Field7C;
```
Comment (blueskythlikesclouds) [inline Sonic/System/GameParameter.h]: Need the p prefix here since it's a pointer.

File: Sonic/System/GameParameter.h
Comment (blueskythlikesclouds) [inline Sonic/System/GameParameter.h:1]: Replace primitive types with ones from cstdint, byte to uint8_t, int to int32_t, etc.

If you are sure that these are structs, you are supposed to remove the m_ prefix from their fields as these are only used for classes.

Also, SStageParameters could be renamed to SStageParameter  for consistency.

File: Sonic/System/GameParameter.h [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+			int m_Field38;
+			int m_Field3C;
+			int m_Field40;
+			BB_INSERT_PADDING(0x18);
+			int m_Field5C;
+			int m_Field60;
+			BB_INSERT_PADDING(0x134);
+			int m_Field198;
+			int m_Field19C;
+			int* m_MissionFlags;
```
Comment (blueskythlikesclouds) [inline Sonic/System/GameParameter.h]: Need the p prefix here since it's a pointer.

**Review by blueskythlikesclouds — CHANGES_REQUESTED** [review]

File: Sonic/System/GameParameter.h (hunk 1 above)
Comment (NextinMono) [reply Sonic/System/GameParameter.h]: Fixed

**Review by blueskythlikesclouds — APPROVED** [review]

**Pull request merged**